Assignment/Assignment/SampleData.cs

```diff
@@ -1,30 +1,100 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.IO;
+using System.Linq;
 
 namespace Assignment
 {
     public class SampleData : ISampleData
     {
+
         // 1.
-        public IEnumerable<string> CsvRows => throw new NotImplementedException();
+        public IEnumerable<string> CsvRows { get; }
 
         // 2.
-        public IEnumerable<string> GetUniqueSortedListOfStatesGivenCsvRows() 
-            => throw new NotImplementedException();
+        public IEnumerable<string> GetUniqueSortedListOfStatesGivenCsvRows()
+        {
+            List<string> states = CsvRows.
+                Select(person => person.Split(',')[6]).
+                Distinct().
+                OrderBy(state => state).
+                ToList();
+
+            return states;
+            //ask preference on where the . belongs on the above. end of row or beginning.
+        }
 
         // 3.
         public string GetAggregateSortedListOfStatesUsingCsvRows()
-            => throw new NotImplementedException();
+        {
+            IEnumerable<string> states = GetUniqueSortedListOfStatesGivenCsvRows();
+            states.ToArray();
+            return string.Join(",", states);
+        }
 
         // 4.
-        public IEnumerable<IPerson> People => throw new NotImplementedException();
+        public IEnumerable<IPerson> People { get; private set; }
 
         // 5.
-        public IEnumerable<(string FirstName, string LastName)> FilterByEmailAddress(
-            Predicate<string> filter) => throw new NotImplementedException();
+        public IEnumerable<(string FirstName, string LastName)> FilterByEmailAddress(Predicate<string> filter)
+        {
+            List<IPerson> persons = People.ToList();
+            IEnumerable<(string FirstName, string LastName)> results = People.ToList().
+                FindAll(person => filter(person.EmailAddress)).
+                Select(person => (person.FirstName, person.LastName));
+
+            return results;
+        }
 
         // 6.
         public string GetAggregateListOfStatesGivenPeopleCollection(
-            IEnumerable<IPerson> people) => throw new NotImplementedException();
+            IEnumerable<IPerson> people)
+        {
+            List<string> states = people
+                .Select(person => (person.Address.State))
+                .Aggregate((states, next) => $"{states},{next}")
+                .Split(",")
+                .Distinct()
+                .OrderBy(state => state)
+                .ToList();
+
+            return string.Join(",", states);

```

Comment:
```suggestion
            return people
                .Select(person => (person.Address.State))
                .Distinct()
                .OrderBy(state => state)
                .Aggregate((states, next) => $"{states},{next}");
```
Instead of aggregating, then splitting, then joining... Simply do all of the selections/ordering/distinct calls then call Aggregate.